client/common/src/types.rs

```diff
@@ -1,4 +1,153 @@
-use trie_db::{Hasher, TrieLayout};
+use std::fmt::Debug;
+
+use codec::decode_from_bytes;
+use codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+use sp_core::Hasher;
+use sp_core::H256;
+use sp_trie::CompactProof;
+use trie_db::TrieLayout;
+
+/// The file chunk size in bytes. This is the size of the leaf nodes in the Merkle tree.
+pub const FILE_CHUNK_SIZE: usize = 1024 * 1024;
 
 /// The hash type of trie node keys
-pub type HashT<T> = <<T as TrieLayout>::Hash as Hasher>::Out;
+pub type HashT<T> = <T as TrieLayout>::Hash;
+pub type HasherOutT<T> = <<T as TrieLayout>::Hash as Hasher>::Out;
+
+// TODO: this is currently a placeholder in order to define Storage interface.
+/// FileKey is the identifier for a file.
+/// Computed as the hash of the FileMetadata.
+pub type Key = Vec<u8>;
+
+// TODO: this is currently a placeholder in order to define Storage interface.
+/// This type mirrors the `FileLocation<T>` type from the runtime, which is a BoundedVec.
+type FileLocation = Vec<u8>;
+
+// TODO: this is currently a placeholder in order to define Storage interface.
+/// Metadata contains information about a file.
+/// Most importantly, the fingerprint which is the root Merkle hash of the file.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub struct Metadata {
+    pub owner: String,
+    pub location: FileLocation,
+    pub size: u64,
+    pub fingerprint: Key,
```

Comment:
Same as Alexandru's comment below, I think setting `pub type Fingerprint = [u8; 32]` for the fingerprint would be better since it's not a file key and we know it's 32 bytes.

Reply:
responded [here](https://github.com/Moonsong-Labs/storage-hub/pull/51#discussion_r1598712585)